prow/cmd/branch-cleaner/branchcleaner_test.go

```diff
@@ -39,6 +39,16 @@ func (f *fakeGithubClient) GetBranches(org, repo string, onlyProtected bool) ([]
 	return branches, nil
 }
 
+func (f *fakeGithubClient) GetBranchProtection(org, repo, branch string) (*github.BranchProtection, error) {
```

Comment:
This never returns an `error`, remove it from the signature?

Reply:
The signature of the method in fake client has to be the same as the one of the regular method ([ref](https://github.com/gardener/ci-infra/blob/ec7d6af0b47b141c12a215440c1390b3fd5911b9/prow/cmd/branch-cleaner/branchcleaner.go#L31), [ref](https://github.com/kubernetes/test-infra/blob/a9938784e8c2501c4ba6518dd5473b95d5dc7580/prow/github/client.go#L168)).
I just added it here because the [upstream Github fake client](https://github.com/kubernetes/test-infra/blob/master/prow/github/fakegithub/fakegithub.go) does not implement this method.

Reply:
Sorry, I missed that this is the fake implementation 🥲